ragaai_catalyst/tracers/utils/rag_trace_json_converter.py

```
@@ -0,0 +1,197 @@
+        
+        else:
+            """Default case for any other GPT models
+            Uses o200k_base encoding as the default tokenizer
+            """
+            return num_tokens_from_string(message, encoding_name="o200k_base")
+        
+
+    # Gemini models 
+    elif re.match(r'^gemini-', model):
+        GOOGLE_API_KEY = os.getenv("GOOGLE_API_KEY")
+        client = genai.Client(api_key=GOOGLE_API_KEY)
```

Ruff (0.8.2)

Undefined name `genai`

(F821)